subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/ImplementationSnapshot.java

```diff
@@ -201,7 +201,7 @@ public boolean isUnknown() {
         @Override
         @Nullable
         public String getUnknownReason() {
-            return "was implemented by a Java lambda";
+            return "was implemented by the Java lambda '" + getTypeName() + "'. Use an anonymous inner class instead.";
```

Comment:
We could add something like "Using Java lambdas is not supported, use an (anonymous) inner class instead."

Reply:
Changed the message.